sdk/core/Azure.Core/tests/HttpPipelineBuilderTest.cs

```diff
@@ -120,6 +120,40 @@ public async Task UsesAssemblyNameAndInformationalVersionForTelemetryPolicySetti
             StringAssert.StartsWith($"azsdk-net-Core.Tests/{informationalVersion} ", value);
         }
 
+        [Test]
+        public async Task UsesAssemblyNameAndInformationalVersionForTelemetryPolicySettingsWithSetTelemetryPackageInfo()
+        {
+            var transport = new MockTransport(new MockResponse(503), new MockResponse(200));
+            var options = new TestOptions
+            {
+                Transport = transport
+            };
+
+            HttpPipeline pipeline = HttpPipelineBuilder.Build(options);
+
+            var message = pipeline.CreateMessage();
+            message.SetUserAgentString(UserAgentValue.FromType<string>());
```

Comment:
nit:  I think it would be more clear if FromType took in a Type type parameter, unless there are some perf considerations here.

I think the usage in our classes would likely be `UserAgentValue.FromType(this.GetType())`

Reply:
Could they use the constructor that takes a `Type` as a parameter instead of changing this method?

Reply:
Good point, @AlexGhiondea - the following ctor is available as well:

https://github.com/Azure/azure-sdk-for-net/blob/4973e11f3b4ac63257fdbc4e4b491ae669954c5f/sdk/core/Azure.Core/src/Pipeline/Internal/UserAgentValue.cs#L23

I think this method is just a bit easier to use.

Reply:
Missed that ctor, yeah we can use that.